src/test/java/org/StudentRepository/StudentRepositoryUnitTest.java

```
@@ -0,0 +1,129 @@
+package org.StudentRepository;
+
+import org.StudentRepository.exceptions.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StudentRepositoryUnitTest {
+
+    private StudentRepository repo;
+    private Student student1;
+    private Student student2;
+
+    @BeforeEach
+    void setUp() {
+        repo = new StudentRepository();
+        try {
+            student1 = new Student("Irina", "Popescu", LocalDate.of(2003, 4, 12), "F", "111");
+        } catch (InvalidStudentException e) {
+            throw new RuntimeException(e);
+        }
+        try {
+            student2 = new Student("Marius", "Pop", LocalDate.of(1997, 2, 26), "M", "222");
+        } catch (InvalidStudentException e) {
+            throw new RuntimeException(e);
+        }
```

if we have used unchecked exceptions, we wouldn't need to do this noisy try-catch-rethrow here. In the end we're not solving anything since we are still throwing RuntimeException and let the program fail. We just added unnecessary code and noise